src/core.js

```diff
@@ -726,13 +726,14 @@ function validateServiceRequiredAttributes(service) {
   }
 
   if (!service.hasOwnProperty('attributes')) {
-    throw Error(`"${service.name}" api has to declare "host", "protocol", and "port" attributes`);
+    throw Error(`"${service.name}" service has to declare "host", "protocol", and "port" attributes or "url" attribute.`);
   }
 
-  if (!service.attributes.hasOwnProperty('port') ||
+  if ((!service.attributes.hasOwnProperty('port') ||
       !service.attributes.hasOwnProperty('host') ||
-      !service.attributes.hasOwnProperty('protocol')) {
-    throw Error(`"${service.name}" api has to declare "host", "protocol", and "port" attributes`);
+      !service.attributes.hasOwnProperty('protocol')) &&
+      !service.attributes.hasOwnProperty('url') ) {
```

Comment:
The `url` property is a write-only shorthand property for host/protocol/port (see [docs](https://docs.konghq.com/0.14.x/admin-api/#request-body)). Our validation was harsher than kong's.